pkg/window/aligned/active.go

```diff
@@ -0,0 +1,98 @@
+// Package aligned maintains the state of active keyed windows in a vertex.
+// Keyed Window maintains the association between set of keys and an interval window.
+// aligned provides functionality Add, Get window for a given interval, remove windows whose End time is before the
+// current watermark.
+package aligned
+
+import (
+	"container/list"
+	"github.com/numaproj/numaflow/pkg/window"
+	"sync"
+	"time"
+)
+
+// ActiveWindows maintains the state of active windows
+// All the operations in ActiveWindows order the entries in the ascending order of start time.
+// So the earliest window is at the front and the oldest window is at the end.
+type ActiveWindows struct {
+	entries *list.List
+	lock    sync.RWMutex
+}
+
+// NewWindows initializes the windows object.
+func NewWindows() *ActiveWindows {
+	return &ActiveWindows{
+		entries: list.New(),
+		lock:    sync.RWMutex{},
+	}
+}
+
+// CreateKeyedWindow adds a new keyed window for a given interval window
+func (aw *ActiveWindows) CreateKeyedWindow(iw *window.IntervalWindow) *KeyedWindow {
+	aw.lock.Lock()
+	defer aw.lock.Unlock()
+
+	kw := NewKeyedWindow(iw)
+
+	// this could be the first window
+	if aw.entries.Len() == 0 {
+		aw.entries.PushFront(kw)
```

Comment:
do we need to run the rest of the code if this is the first entry?

Reply:
addressed.